firmware/common/sonde_packet.cpp

```diff
@@ -109,6 +111,12 @@ GPS_data Packet::get_GPS_data() const
 		result.lat = reader_bi_m.read(14 * 8, 32) / ((1ULL << 32) / 360.0);
 		result.lon = reader_bi_m.read(18 * 8, 32) / ((1ULL << 32) / 360.0);
 	}
+	else if (type_ == Type::Meteomodem_M20)
+	{
+		result.alt = ((reader_bi_m.read(8 * 8, 32) / 100) - 48) / 250 ; //Conversion of altitude is actually a bit shifted, needs to be more accurate.
```

Comment:
reading the linked decoder code, the translation would be:
result.alt = reader_bi_m.read(8 * 8, 24) / 100 ;  

There is a constant -48 on M10 and M2K2. Not sure if needed here, but if it is then:

result.alt = (reader_bi_m.read(8 * 8, 24) / 100) - 48 ;

Reply:
What I changed from your code is that I only read 3 bytes as the alt is encoded on 3 bytes only (3 * 8  =24)
I dunno if it is compiling and I have no sonde to check.

Reply:
From what I've read further, just combining the 3 bytes and divide by 10 should do it